applications/GeoMechanicsApplication/custom_constitutive/interface_coulomb_with_tension_cut_off.cpp

```diff
@@ -0,0 +1,219 @@
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//
+//  Main authors:    Mohamed Nabi,
+//                   Wijtze Pieter Kikstra
+//
+
+// Application includes
+#include "custom_constitutive/interface_coulomb_with_tension_cut_off.h"
+#include "custom_constitutive/constitutive_law_dimension.h"
+#include "custom_utilities/constitutive_law_utilities.h"
+#include "custom_utilities/math_utilities.h"
+#include "custom_utilities/stress_strain_utilities.h"
+#include "geo_mechanics_application_constants.h"
+#include "geo_mechanics_application_variables.h"
+
+#include <cmath>
+#include <optional>
+
+namespace
+{
+
+using namespace Kratos;
+
+void CheckProperty(const Properties&       rMaterialProperties,
+                   const Variable<double>& rVariable,
+                   std::optional<double>   MaxValue = std::nullopt)
+{
+    KRATOS_ERROR_IF_NOT(rMaterialProperties.Has(rVariable))
+        << rVariable.Name() << " is not defined for property " << rMaterialProperties.Id() << std::endl;
+    KRATOS_ERROR_IF(rMaterialProperties[rVariable] < 0.0 ||
+                    (MaxValue.has_value() && rMaterialProperties[rVariable] > MaxValue.value()))
+        << "value of " << rVariable.Name() << " for property " << rMaterialProperties.Id()
+        << " is out of range: " << rMaterialProperties[rVariable] << " is not in [0.0, "
+        << (MaxValue ? std::to_string(*MaxValue) + "]" : "->") << std::endl;
+}
+
+} // namespace
+
+namespace Kratos
+{
+ConstitutiveLaw::Pointer InterfaceCoulombWithTensionCutOff::Clone() const
+{
+    auto p_result                      = std::make_shared<InterfaceCoulombWithTensionCutOff>(*this);
+    p_result->mTractionVector          = mTractionVector;
+    p_result->mTractionVectorFinalized = mTractionVectorFinalized;
+    p_result->mRelativeDisplacementVectorFinalized = mRelativeDisplacementVectorFinalized;
+    p_result->mCoulombWithTensionCutOffImpl        = mCoulombWithTensionCutOffImpl;
+    return p_result;
+}
+
+Vector& InterfaceCoulombWithTensionCutOff::GetValue(const Variable<Vector>& rVariable, Vector& rValue)
+{
+    if (rVariable == CAUCHY_STRESS_VECTOR) {
+        rValue = mTractionVector;
+    } else {
+        rValue = ConstitutiveLaw::GetValue(rVariable, rValue);
+    }
+    return rValue;
+}
+
+void InterfaceCoulombWithTensionCutOff::SetValue(const Variable<Vector>& rVariable,
+                                                 const Vector&           rValue,
+                                                 const ProcessInfo&      rCurrentProcessInfo)
+{
+    if (rVariable == CAUCHY_STRESS_VECTOR) {
+        mTractionVector = rValue;
+    } else {
+        KRATOS_ERROR << "Can't set value of " << rVariable.Name() << ": unsupported variable\n";
+    }
+}
+
+SizeType InterfaceCoulombWithTensionCutOff::WorkingSpaceDimension() { return N_DIM_2D; }
```

Comment:
Let's add a comment here:
```suggestion
SizeType InterfaceCoulombWithTensionCutOff::WorkingSpaceDimension()
{
    // Note that this implementation assumes line interface elements. It needs to be modified when planar interface elements become available.
    return N_DIM_2D;
}
```